src/AlgebraicGeometry/ToricVarieties/ToricLineBundles/cech-cohomology.jl

```diff
@@ -0,0 +1,101 @@
+#The following function is called, when algorithm="chambers" is specified for all_cohomologies
+function _all_cohomologies_via_cech(tl::ToricLineBundle)
+  our_maps = transpose.(matrix.(_toric_cech_complex(tl)))
+  X = toric_variety(tl)
+  alternating_sum = abs(sum(binomial(n_maximal_cones(X), k) * (-1)^k for k in dim(X)+2:n_maximal_cones(X); init = 0))
+  alternating_sum *= Int(ncols(our_maps[end]) // binomial(n_maximal_cones(X), dim(X)+1))
+  @req ncols(our_maps[end]) >= alternating_sum "Inconsistency encountered"
+  rks_maps = push!(rank.(our_maps), alternating_sum)
+  return ZZRingElem.([nrows(our_maps[1]) - rks_maps[1]; [nrows(our_maps[k]) - rks_maps[k] - rks_maps[k-1] for k in 2:dim(X)]; ncols(our_maps[dim(X)]) - rks_maps[dim(X) + 1] - rks_maps[dim(X)]])
+end
+
+function _toric_cech_complex(tl::ToricLineBundle)
+  # Extract essential information
+  X = toric_variety(tl)
+  @req is_complete(X) "Toric variety must be complete!"
+
+  # Compute the hyperplane arrangement
+  a_plane = matrix(ZZ, n_rays(X), 1, coefficients(toric_divisor(tl)))
+  sc = cone_from_inequalities(matrix(ZZ, [[-1; zeros(Int, dim(X))]]))
+  H = Polymake.fan.HyperplaneArrangement( HYPERPLANES = [a_plane matrix(ZZ, rays(X))], SUPPORT=sc.pm_cone)
+
+  # Classify the regions and their lattice points that contribute
+  pff = Polymake.fan.PolyhedralComplex(POINTS=H.CHAMBER_DECOMPOSITION.RAYS, INPUT_CONES=H.CHAMBER_DECOMPOSITION.MAXIMAL_CONES)
+  max_polys = maximal_polyhedra(polyhedral_complex(pff))
+  other_polys = vcat([polyhedra_of_dim(polyhedral_complex(pff), i) for i in 0:dim(polyhedral_complex(pff))-1]...)
+  chamber_signs = matrix(ZZ, H.CHAMBER_SIGNATURES)
+  sign_of_chamber = Dict(chamber_signs[i,:] => interior_lattice_points(p) for (i, p) in enumerate(max_polys) if is_bounded(p))
+  list_of_boundary_lattice_points = vcat([boundary_lattice_points(p) for p in values(filter(is_bounded, max_polys))]...)
+  append!(list_of_boundary_lattice_points, vcat([lattice_points(p) for p in other_polys if is_bounded(p)]...))
+  list_of_boundary_lattice_points = unique(list_of_boundary_lattice_points)
+  sign_of_boundary_pt = Dict(pt => matrix(QQ, rays(X)) * pt + a_plane for pt in list_of_boundary_lattice_points)
+
+  # Prepare information, which we use to iterate over the Cech complex and identify the relevant lattice points
+  RI = ray_indices(maximal_cones(X))
+  ray_index_list = map(row -> findall(!iszero, collect(row)), eachrow(RI))
+
+  # Now iterate over the Cech complex
+  cech_complex_points = Vector{Dict{Vector{Int64}, Vector{PointVector{ZZRingElem}}}}(undef, dim(X) + 1)
+  #cech_complex_maps = Vector{SMat{ZZRingElem, Hecke.ZZRingElem_Array_Mod.ZZRingElem_Array}}(undef, dim(X))
+  cech_complex_maps = Vector{Any}(undef, dim(X))
```

Comment:
```suggestion
  cech_complex_maps = Vector{ModuleFPHom}(undef, dim(X))
```
? 

Or maybe even more specific? I'm also a bit puzzled about the range being bounded by `dim(X)`. Usually, a Cech complex is longer than that, isn't it?

Reply:
Here for now we only compute the maps up to `dim(X)` and then use the fact that the tail is exact to compute cohomology dimensions.